eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,160 @@
+function Get-java-OnboardedDocsMsPackages($DocRepoLocation) {
+    $packageOnboardingFiles = "$DocRepoLocation/package.json"
+  
+    $onboardingSpec = ConvertFrom-Json (Get-Content $packageOnboardingFiles -Raw)
+    $allPackages = @{}
+    foreach ($spec in $onboardingSpec) {
+      $spec.packages | ForEach-Object {$allPackages[$_.packageArtifactId] = $null}
+    }
+    return $allPackages
+  }
+  
+  function Get-java-DocsMsTocData($packageMetadata, $docRepoLocation) {
+    # Fallback to get package-level readme name if metadata file info does not exist
+    $packageLevelReadmeName = $packageMetadata.Package.Replace('azure-', '');
+  
+    # If there is a metadata json for the package use the DocsMsReadmeName from
+    # the metadata function
+    if ($packageMetadata.PSObject.Members.Name -contains "FileMetadata") {
+      $readmeMetadata = &$GetDocsMsMetadataForPackageFn -PackageInfo $packageMetadata.FileMetadata
+      $packageLevelReadmeName = $readmeMetadata.DocsMsReadMeName
+    }
+  
+    $packageTocHeader = $packageMetadata.Package
+    if ($clientPackage.DisplayName) {
```

Comment:
If the above FileMetadata check needs to access PSObject due to strict mode property not found exceptions, should that also happen here or is DisplayName guaranteed to exist?

Reply:
Good catch

Reply:
Double check the data. The displayName guaranteed to exist. It is from one of the column in this table https://github.com/Azure/azure-sdk/blob/main/_data/releases/latest/java-packages.csv

FileMetadata is newly added member through code flow.